chain/store/stateGenerator.go

```diff
@@ -263,7 +286,7 @@ func (cs *ChainStore) generateStateRoot(ctx context.Context, b *block.Block, gen
 					return nil, common.EmptyUint256, err
 				}
 
-				if err = states.UpdateID(pg[0], id); err != nil {
+				if err = states.UpdateID2(pg[0], id); err != nil {
```

Comment:
For id 2 this should be the pg of txn public key, NOT from txn.GetProgramHashes()